app.rb

```diff
@@ -122,35 +132,36 @@
     event_time = Time.parse("#{monitor['last_event']['utc_timestamp']}Z")
   end
   
-  local_event_id = DB["monitor_events"].insert({
+  local_event = MonitorEvent.create(
     state: state_abbreviation,
     monitor: params[:monitor_id],
     status: event_status,
     date: event_time,
     pingometer_id: event_id
-  })
+  )
   
   # Update incidents
-  last_incident = DB["incidents"].find({monitor: params[:monitor_id]}).sort({start_date: -1}).first
+  last_incident = Incident.where(monitor: params[:monitor_id]).sort({start_date: -1}).first
   if event_status == 0
-    if last_incident && last_incident["end_date"].nil?
-      last_incident["events"] << local_event_id
-      DB["incidents"].update({"_id" => last_incident["_id"]}, last_incident)
+    if last_incident && last_incident.end_date.nil?
```

Comment:
I think `ongoing?` makes more sense. The question isn’t really whether there’s an end date, but whether the incident is still happening.